### PR TITLE
Address compiler warnings

### PR DESCRIPTION
**Pull request by funkyboy**
See here: https://github.com/ably/ably-ios/issues/689

### CODE REVIEW

**Comment by funkyboy** [comment]
@ricardopereira @rvi ready for review

File: Pod/Classes/MessagePackParser+Streaming.h
Comment (ricardopereira) [inline Pod/Classes/MessagePackParser+Streaming.h:15]: Isn't this a breaking change?

Comment (funkyboy) [reply Pod/Classes/MessagePackParser+Streaming.h:15]: @ricardopereira This project compiles. 

And we are [not using it in ably-iOS](https://github.com/ably/ably-ios/search?utf8=✓&q=initWithBufferSize&type=)

If you want to keep it because other projects are using it, let me know.

Comment (ricardopereira) [reply Pod/Classes/MessagePackParser+Streaming.h:15]: I'm good with that change. I defer to @rvi 😄

**Review by rvi — APPROVED** [review]

**Comment by ricardopereira** [comment]
@rvi Could you please release a new pod version with this changes. Thanks

**Comment by rvi** [comment]
Hey, I spent 2h trying to build it but unfortunately, I'm not able to do it. The `pod lib lint` command doesn't work, can't find the umbrella header... I'll try to work on it next week if I find the time. Feel free to help :)

**Comment by funkyboy** [comment]
@rvi @ricardopereira I made it work but you have to:

- roll back to remove new project settings (https://github.com/rvi/msgpack-objective-C/commit/e9f7b144b9a06deee15fc55d36e8b7be6b919509)
- install Xcode 8.3
- use old version of Cocoapods `gem install cocoapods -v 0.37.2`

Is this bananas? Yes, but after two hours of investigation I could not find a better solution :)

**Comment by ricardopereira** [comment]
@funkyboy Sorry, I didn't get it. Why using Xcode 9 isn't working?

**Comment by funkyboy** [comment]
@ricardopereira You didn't get it, because I didn't tell you. I didn't tell you because I don't know :)
Error is "can't find msgpack-umbrella2.h", which is declared in the modulemap.

If you want to take a stab at it, feel free. I suspect it doesn't have a lot to do with Xcode, but with the changes to Cocoapods. Last version of this pod was released with `0.37.2` and now we are at `1.4.0`.

If you can figure out what we need to change, let me know.

**Comment by ricardopereira** [comment]
@funkyboy Ah, ok. CocoaPods sometimes can give you big headaches 😄 .

**Comment by ricardopereira** [comment]
Did you tried removing this line? https://github.com/rvi/msgpack-objective-C/blob/0ca280a06024baf2adaae3256b0b116138985d54/msgPack.modulemap#L3

**Comment by funkyboy** [comment]
@ricardopereira that's the first thing I tried :)

**Comment by ricardopereira** [comment]
@funkyboy You removed it but the `pod lib lint` will always pull the code from tag `0.2.0` (current podspec).

@rvi Can you give me write permissions to this repo to help you out?

**Comment by funkyboy** [comment]
@ricardopereira I did my thing from the fork here: https://github.com/ably-forks/msgpack-objective-C
Apart from the version it should be the same.

**Comment by rvi** [comment]
@ricardopereira Sure, but why not a pull request, it's easier to read.

@funkyboy I can give you cocoapods push access if you want, will be easier, I'm in a small Mexican fishing village with a very bad network, won't be possible to download Xcode 8.3. Just send me your cocoapods trunk login email (my email is: firstname@lastname.us) you probably don't remember, but we spoke at the same conference, MCE first version in 2014 ^^

**Comment by funkyboy** [comment]
@rvi did we? I barely remember what I had for breakfast :) 
Thanks for helping with this. I sent you an email but it bounced back. Here's mine cesare AT ably.io. Will send a PR tomorrow.
Enjoying fishing (or eating fish :)

**Comment by rvi** [comment]
@funkyboy Done.

Not fishing, but kiteboarding, even better 🤙

**Pull request merged**
